radar/exporter/exporters.py

```diff
@@ -357,96 +357,164 @@ def setup(self):
+                            self.ins_data['ins_dia_date'] = ''
+                            row[29] = self.ins_data['ins_biopsy_dia']
+                            self.ins_data['ins_biopsy_dia'] = ''
+                            row[30] = self.ins_data['ins_biopsy_dia_label']
+                            self.ins_data['ins_biopsy_dia_label'] = ''
+                            row[31] = self.ins_data['ins_biopsy_dia_comments']
+                            self.ins_data['ins_biopsy_dia_comments'] = ''                   
+                            yield row
+                            row = self.makeRow(result, diagnosis)
+                            ins_data = self.getInsData(result, diagnosis)
+                        # New patient, no data found for previous patient
+                        else:
+                            row = self.makeRow(result, diagnosis)
+                            ins_data = self.getInsData(result, diagnosis)
+                
+                # Not new patient
+                else:
+                    # Primary diagnosis has been found, continue to check for INS data
+                    if row:
+                        ins_data = self.getInsData(result, diagnosis)
+                    # Search for primary diagnosis and INS data
+                    else:
+                        row = self.makeRow(result, diagnosis)
+                        ins_data = self.getInsData(result, diagnosis)
+        
+        # Not Nurture-ins export
+        else:            
+            for result in self._query:
+                diagnosis = result.diagnosis
+                if not diagnosis:
+                    continue
+                row = self.makeRow(result, diagnosis)
+                if row:
+                    yield row  
+
+    def makeRow(self, result, diagnosis):        
         
         if diagnosis not in self._primary:
             return
-        
-        self.row = [col[1](result) for col in self._columns]
-        
-        if diagnosis and diagnosis.codes:
-            for code in diagnosis.codes:
-                if code.system == "ERA-EDTA PRD":
-                    self.row[5] = code.code
-                if code.system == "ICD-10":
-                    self.row[6] = code.code
-                if code.system == "SNOMED CT":
-                    self.row[7] = code.code
+        else:
+            row = [col[1](result) for col in self._columns]
+            
+            if diagnosis and diagnosis.codes:
+                for code in diagnosis.codes:
+                    if code.system == "ERA-EDTA PRD":
+                        row[5] = code.code                        
+                    if code.system == "ICD-10":
+                        row[6] = code.code                        
+                    if code.system == "SNOMED CT":
+                        row[7] = code.code
+
+            return row
+
+    def getInsData(self, result, diagnosis):
```

Comment:
It would be helpful if this could have a comment saying what "result" is. I expected it to be a test result from the name but it's not.

Reply:
Result is a result from the query and is in keeping with the pre-existing code. I can add a note though to explain what a result is. I have added a comment here as to where I think the comment should go